craft_application/services/lifecycle.py

```diff
@@ -106,18 +107,20 @@ class LifecycleService(base.BaseService):
         LifecycleManager on initialisation.
     """
 
-    def __init__(
+    def __init__(  # noqa: PLR0913 (too many arguments)
         self,
         app: AppMetadata,
         project: Project,
         *,
         work_dir: Path | str,
         cache_dir: Path | str,
+        build_for: str,
```

Comment:
add this to the class docstring please

Reply:
Added a description for `build_for` and for `cache_dir`, which was also missing.